### DISCUSSION TITLE
Next JS route internationalization

### DISCUSSION THREAD

**Discussion by emillind** [vercel/next.js#31299] [Help] [answered]
Hi! 

Me and my team are running into some question marks when we are trying to update the page structure for our website. We have previously used a very dynamic way of rendering our pages, essentially having a `pages/[slug]` that dynamically resolves what type of page we should render. This solution comes from using one build job to build pages for multiple locales in multiple markets.

However, we have recently moved to use a monorepo to separate our different sites by market. Therefore we don't have the need for the same level of dynamic routes but we still want to them to be internationalized, without having to duplicate code.

Essentially the same content should be available on different languages on:
* `/platser/[region]/[slug]`          - swedish route on swedish market
* `/en/locations/[region]/[slug]` - english route on swedish market

The `/en/` part we have used the `i18n` property in the next config for but we are looking for a way to solve the platser/locations part. Ideally we would have a folder structure like so: `pages/platser/[region]/[slug].tsx` that would resolve both the paths mentioned above.

Basically what i am looking for is a way of doing this without having to have a `/pages/locations` folder, thus not having to duplicate code.

Any ideas, potential problems or similar?

**Comment by kimgiutae** [comment, marked as answer]
Hi. Maybe related to https://github.com/vercel/next.js/discussions/18485, https://github.com/vercel/next.js/discussions/18572 and https://github.com/vercel/next.js/discussions/31178.

Seems that the devs offers a way to locale all pages but without a native way to translate page names (url itself). Community comes with some workarounds but sometimes works and sometimes not.

If that is your case, please upvote and comment the others issues I cite above to make the devs aware of this issue/feature.
